lib/uuid.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+uuid_fallback()
+{
+    local N B C='89ab'
+
+    for (( N=0; N < 16; ++N ))
+    do
+        B=$(( RANDOM%256 ))
+
+        case $N in
+            6)
+                printf '4%x' $(( B%16 ))
+                ;;
+            8)
+                printf '%c%x' ${C:$RANDOM%${#C}:1} $(( B%16 ))
+                ;;
+            3 | 5 | 7 | 9)
+                printf '%02x-' $B
+                ;;
+            *)
+                printf '%02x' $B
+                ;;
+        esac
+    done
+
+    echo
+}
+
+uuid() {
```

Comment:
Since Ruby is available on macOS, and the stack image, can we use that? Something like `ruby -e "require 'securerandom'; puts SecureRandom.uuid"` should work and give us actual GUIDs so that we don't have to rely on a fallback.

Reply:
🤔  People use this buildpack in other environments that we don't control, and I'd prefer not to add ruby (or another runtime) as a dependency just for this. It's also unlikely that those users are going to be using the log output.

I know other buildpacks call out to ruby or python successfully, but I don't think this warrants that big of a change.

Reply:
I'm going to be embedding small go binaries for this kind of thing in the buildpack soon, so maybe I should revisit this then

Reply:
So, that context takes this from "We could have a lot less guessing about things" to "I don't think we should provide the UUID fallback".  As you've pointed out, people use this in other environments and might pull this in function in where the main options are missing. Is the bash fallback secure or going to generate a truly unique UUID?  It's hard for me to tell and I don't know what guarantees can be made around `$RANDOM`

Reply:
> Is the bash fallback secure or going to generate a truly unique UUID? It's hard for me to tell and I don't know what guarantees can be made around `$RANDOM`

Nor do I. I originally had it emitting a constant "0000-0000" type string, but that breaks tests. I could have it emit an empty string and then cover that case 🤔

Reply:
I just don't want it to crash on Dokku users just because it's trying to pipe something to `/dev/null`

Reply:
Might just be good to add a comment explaining why we need a fallback like that and the fact that we can't guarantee uniqueness for realz.

Reply:
That I can do 👍